met-api/migrations/versions/a3e6dae331ab_add_visibility_column_to_engagment.py

```diff
@@ -0,0 +1,63 @@
+"""Add visibility column to engagment
+
+Revision ID: a3e6dae331ab
+Revises: 9a93fda677eb
+Create Date: 2024-05-28 08:26:11.155679
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = 'a3e6dae331ab'
+down_revision = '9a93fda677eb'
+branch_labels = None
+depends_on = None
+
+def upgrade():
+    # Create the engagement_visibility table
+    engagement_visiblity_table = op.create_table(
+        'engagement_visibility',
+        sa.Column('id', sa.Integer(), nullable=False, autoincrement=True),
+        sa.Column('visibility_name', sa.String(length=50), nullable=False),
+        sa.Column('description', sa.String(length=100), nullable=False),
+        sa.Column('created_date', sa.TIMESTAMP(timezone=False), nullable=False, server_default=sa.text('CURRENT_TIMESTAMP')),
+        sa.Column('updated_date', sa.TIMESTAMP(timezone=False), nullable=True),
+        sa.PrimaryKeyConstraint('id')
+    )
+    # Insert the initial data into the engagement_visibility table
+    op.bulk_insert(
+        engagement_visiblity_table,
+        [
+            {'id': 1, 'visibility_name': 'Public', 'description': 'Visible to all users'},
+            {'id': 2, 'visibility_name': 'Slug', 'description': 'Accessible to users with the direct link'},
+            {'id': 3, 'visibility_name': 'AuthToken', 'description': 'Visible to authenticated users'}
```

Comment:
Should these be made into a constant like in met-api\src\met_api\constants

Reply:
I did add these as a constants to `met-api/src/met_api/constants/engagement_visibility.py`. I based this off of how the engagement_status table was added, but would it be better to reference these constants in the migration file?

Reply:
Oh i see that now. One thing you could do when adding the visibilty column to the engagement table is rather than giving it type sa.Integer() you could do sa.Enum(Visibility) so its restricted to those integers and not all integers.

Reply:
It is restricted to those integers because of the foreign key constraint. 
There are a number of enums used in other tables, but there also seem to be foreign key constraints used instead to reference tables (like engagement_status) rather than an enum... 
hmm yea not sure if one is better than the other.